src/dmd/root/speller.d

```diff
@@ -95,83 +95,82 @@ private auto spellerY(alias dg)(const(char)[] seed, size_t index, ref int cost)
     return p; // return "best" result
 }
 
-private auto spellerX(alias dg)(const(char)* seed, size_t seedlen, bool flag)
+private auto spellerX(alias dg)(const(char)[] seed, bool flag)
 {
-    if (!seedlen)
+    if (!seed.length)
         return null;
     char[30] tmp;
-    char* buf;
-    if (seedlen <= tmp.sizeof - 2)
-        buf = tmp.ptr;
+    char[] buf;
+    if (seed.length <= tmp.sizeof - 1)
```

Comment:
Why is this `-1` as opposed to `-2`?

Reply:
Space for the terminating null. No longer needed.